### ISSUE TITLE
struct.error: unpack requires a buffer of 2 bytes

### ISSUE THREAD

**Issue by josdegroot** [boredazfcuk/docker-icloudpd#125]
I'm running iCloudPD on a Synology NAS within docker, but tried everything but get the following error.

I use the following variables

<img width="883" alt="Screenshot 2022-01-31 at 22 52 45" src="https://user-images.githubusercontent.com/77342255/151879346-a252360f-cc2a-48bc-810f-6a1824fb8346.png">

date | stream | content
-- | -- | --
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Next synchronisation at 15:42:29
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Total time taken: 01:31:37
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Synchronisation ended at 17:14:06
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     iCloud login counter = 2
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Days remaining until expiration: 83
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Two factor authentication cookie expires: 2022-04-24 @ 23:40:09
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 INFO     Web cookie expires: 2022-03-25 @ 23:39:53
2022-01-31 16:14:06 | stdout | 2022-01-31 17:14:06 ERROR    Error during download - Exit code: 1
2022-01-31 16:14:06 | stdout | struct.error: unpack requires a buffer of 2 bytes
2022-01-31 16:14:06 | stdout | tag_count = struct.unpack(self.endian_mark + "H",
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/piexif/_load.py", line 108, in get_ifd_dict
2022-01-31 16:14:06 | stdout | exif_dict["Interop"] = exifReader.get_ifd_dict(pointer, "Interop")
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/piexif/_load.py", line 49, in load
2022-01-31 16:14:06 | stdout | exif_dict = piexif.load(path)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/exif_datetime.py", line 11, in get_photo_exif
2022-01-31 16:14:06 | stdout | (".jpg", ".jpeg")) and not exif_datetime.get_photo_exif(download_path):
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/base.py", line 489, in download_photo
2022-01-31 16:14:06 | stdout | download_photo(consecutive_files_found, item)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/icloudpd-1.7.2-py3.8.egg/icloudpd/base.py", line 562, in main
2022-01-31 16:14:06 | stdout | return callback(*args, **kwargs)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/click/core.py", line 535, in invoke
2022-01-31 16:14:06 | stdout | return ctx.invoke(self.callback, **ctx.params)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/click/core.py", line 895, in invoke
2022-01-31 16:14:06 | stdout | rv = self.invoke(ctx)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/click/core.py", line 697, in main
2022-01-31 16:14:06 | stdout | return self.main(*args, **kwargs)
2022-01-31 16:14:06 | stdout | File "/usr/lib/python3.8/site-packages/click/core.py", line 722, in __call__
2022-01-31 16:14:06 | stdout | sys.exit(load_entry_point('icloudpd==1.7.2', 'console_scripts', 'icloudpd')())
2022-01-31 16:14:06 | stdout | File "/usr/bin/icloudpd", line 33, in <module>
2022-01-31 16:14:06 | stdout | Traceback (most recent call last):

**Comment by boredazfcuk** [comment]
Hi,

Can you try unsetting the **set_exif_datetime** variable, and re-downloading?

I'm not a Python developer so don't really understand what the underlying application is complaining about, but it seems to be related to exif stuff.

It might be that there is a photo in your iCloud which has corrupted exif data in it, or just some special character that Python isn't expecting and can't handle.

Let me know if that successfully downloads your photos, if it does, then it would be great if you can work out which image is causing the problem. However, that could be a monumental task if you have tens of thousands of photos in there.

**Comment by josdegroot** [comment]
It's syncing now... I have more then 80K images stored in iCloud at the moment so it would be impossible to find the image that maybe had corrupted exif data. :).

Keep you posted if it works with the set_exif_datetime on false...

Can you maybe explain what will be the difference now with this setting on false?

**Comment by boredazfcuk** [comment]
Hi, the **set_exif_datetime** variable enables the `--set-exif-datetime` command line option in the underlying application ([icloud_photo_Downloader](https://github.com/icloud-photos-downloader/icloud_photos_downloader))

With this set, if the appliction detects that the photo's DateTimeOriginal EXIF tag does not have a value, it reads the file's creation time and uses that to set the EXIF tag.

Without it, the DateTimeOriginal EXIF tag will remain empty. However, I've found that enabling the setting means that the downloaded file differs from the file which is in icloud, so it will end up duplicating the file.

**Comment by boredazfcuk** [comment]
Cool. I've actually switched the feature off on my containers since you created this issue.

I looked at all the files that were having this EXIF tage set, and they are all just stupid memes and other JPGs I've saved to my photo stream, to then WhatsApp my idiot friends.

It causes the downloader to de-duplicate the files, because the ones with the updated EXIF tags are different from the ones in the cloud. Means I had two copies of them instead of one. These de-duplicated files remain after you delete the original from your phone. It will delete the original from the downloaded files, but not the de-duped one, so I also had to delete out a cload of clutter too. More trouble than it's worth I think.
